src/main/java/org/reaktivity/k3po/nukleus/ext/internal/behavior/NukleusStreamFactory.java

```diff
@@ -167,8 +169,10 @@ private void onData(
                 {
                     partition.doWindow(channel, readableBytes, 1);
                 }
-
-                fireMessageReceived(channel, message);
+                if (data.authorization() == authorization)
```

Comment:
`==` or mask check?

Reply:
Yes, should be mask check. Thanks for catching this. Test should've been failing but there was another error, which I have now fixed: authorization the checking against should be that of the server channel, not that set on the begin frame from the client.